### ISSUE TITLE
html.elements.del - Safari with VoiceOver does not announce text preceding `<del>` element

### ISSUE THREAD

**Issue by aardrian** [mdn/browser-compat-data#27443]
### What type of issue is this?

Incorrect support data.

### What information was incorrect, unhelpful, or incomplete?

Support for the `<del>` element is incomplete / has regressed in Safari 18 (through 18.5 as of this writing). When paired with VoiceOver it functionally removes preceding content from the page.

### What browsers does this problem apply to, if applicable?

Safari

### What did you expect to see?

When using VoiceOver with Safari, that the text preceding a `<del>` would be announced.

### Did you test this? If so, how?

In a test using `<del>` or `role="deletion"`, VoiceOver with Safari does not announce the opening of the sentence, instead jumping to where the element starts.

To replicate:

1. Open Safari and activate VoiceOver
2. Go to:
   * https://cdpn.io/aardrian/debug/rNRxZmP
   * https://adrianroselli.com/demos/text-level-semantics-reprised/
3. Use the `read-all` command (VO + A)
4. Observe how, for most paragraphs, the words preceding the element are not announced.

### Can you link to any release notes, bugs, pull requests, or MDN pages related to this?

[286267: AX: VoiceOver does not announce text preceding some phrasing elements](https://bugs.webkit.org/show_bug.cgi?id=286267)

### Do you have anything more you want to share?

https://github.com/user-attachments/assets/0ec18f45-a612-4e14-a7a8-922d9c488162

### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/del

### MDN metadata

<details>
<summary>MDN page report details</summary>

* Query: `html.elements.del`
* Report started: 2025-07-29T20:07:02.312Z

</details>

**Comment by aardrian** [comment]
Related:
* https://github.com/mdn/browser-compat-data/issues/27443
* https://github.com/mdn/browser-compat-data/issues/27444
* https://github.com/mdn/browser-compat-data/issues/27445
* https://github.com/mdn/browser-compat-data/issues/27446

**Comment by caugner** [comment]
Thank you for taking the time to report these issues, and open corresponding PRs. 🙌 

However, screen reader compatibility is out of scope for the BCD project, see the [What isn't tracked](https://github.com/mdn/browser-compat-data#what-isnt-tracked) section in the README. So I'm going to close these issues and PRs as "won't fix".

Thank you for your understanding. 🙏

**Comment by aardrian** [comment]
FWIW, this is not about screen reader compat, this is a bug in how Safari works. That it happens to be obvious with screen readers is little different than if the text was visually hidden or displayed in a different font. It's the browser messing it up.

So no, I don't understand.

**Comment by caugner** [comment]
@aardrian I'm open to revisit the decision.

The HTML Standard specifies the `<del>` element in [this section](https://html.spec.whatwg.org/multipage/edits.html#the-del-element). Can you point out where the HTML Standard (or any another specification) mentions the behavior that has regressed in Safari 18?

**Comment by aardrian** [comment]
@caugner Just so I am clear, do you consider it conformant with the HTML5 specification for the element to trigger a behavior that hides preceding content?

**Comment by ddbeck** [comment]
There are few things about this (and the related bug reports) that are unclear to me:

- Is it possible to establish if this is a bug of Safari or a bug of VoiceOver? For example, does using VoiceOver with Chrome or Firefox produce similar results? Does an alternate screen reader used with Safari (if one exists) fail in the same way? Or does a braille display work, but audio does not?
- Does this bug happen in iOS, macOS, or both?

For comparison here, if a similar bug affected Chrome with JAWS but did not affect Chrome with NVDA or Chrome with VoiceOver, then we'd have evidence that it's a problem with the screen reader software and not the browser (thus out of scope for BCD). On the other hand, if a similar bug affected all screen readers, we'd have evidence that it's a browser bug and it would potentially be in-scope (setting aside some exceptions that wouldn't seem to apply here). As originally reported, I cannot guess as to which situation applies here.

**Comment by aardrian** [comment]
@ddbeck 

> * Is it possible to establish if this is a bug of Safari or a bug of VoiceOver?

Not conclusively without access to the code for VoiceOver and the macOS AAPIs and also having the programming skills to wade into the WebKit code. If you have that access and those skills, I would be delighted if you could confirm it conclusively. Alternatively, if you have a contact at Apple who can explore the bug (so far still marked as "new" and with no comments from Apple) and provide a conclusive answer, that would be swell.

> * For example, does using VoiceOver with Chrome or Firefox produce similar results?

No. See the WebKit bug https://bugs.webkit.org/show_bug.cgi?id=286267 (also linked in this issue).

> * Does an alternate screen reader used with Safari (if one exists) fail in the same way?

There is no other screen reader on macOS (there are experimental third-party efforts, but I understand the macOS AAPIs are not documented).

> * Or does a braille display work, but audio does not?

Yes (emulator only; unable to afford a Braille display). See the WebKit bug https://bugs.webkit.org/show_bug.cgi?id=286267 (also linked in this issue).

> * Does this bug happen in iOS, macOS, or both?

macOS. See the WebKit bug https://bugs.webkit.org/show_bug.cgi?id=286267 (also linked in this issue).

**Comment by ddbeck** [comment]
> > * Or does a braille display work, but audio does not?
> 
> Yes (emulator only; unable to afford a Braille display). See the WebKit bug https://bugs.webkit.org/show_bug.cgi?id=286267 (also linked in this issue).

Ah, OK. Thank you. This text then:

> * The unannounced text appears in the speech viewer.
> * The unannounced text appears in the Braille viewer.

This suggests to me that Safari is properly exposing the text to the AT, but the AT is failing to produce the desired audio. As you say, we cannot know for certain what code is causing this, but right now the evidence we have does not yet confirm that this is a bug of WebKit/Safari.

My suggestion here is that when/if the WebKit bug is confirmed (and/or some fix appears as a PR on webkit/webkit), then we should reopen this and figure out the best way to cover it in BCD. I don't have any special contacts at Apple that might hasten this, unfortunately.

**Comment by aardrian** [comment]
IMO, there is more than enough evidence to tie it to the browser, given the issue I cited _only_ happens in Safari, even when other aspects are exposed. It's still a bug.

The challenge here is that we can't be absolutely completely certain there isn't something in the macOS AAPI or in VoiceOver that only gets triggered when it's listening to Safari and only Safari. Apple dragging its feet to confirm it means this can live forever unreported / untracked (which is to Apple's benefit). But with the evidence we _have_, and the evidence we don't, the issue is clearly exposed _only_ in Safari. So, Safari bug.
